### PR TITLE
Feature/6 understanding golang exports

### LINKED ISSUE #6: understanding golang exports
1. Initialize go module on the lesson repository. Use `go mod init` command to initialize it.
2. Create 50 packages.
3. On each packages, export a public variable and a function.
4. On each packages, there must be a private variable and a function which are not exported.
5. The exported variables of each packages must be used in the main package.
6. The private variable and function of each packages must be used within each packages.
7. Commit your changes after completing each package. (Total commits should not be lesser than 50)

### PR DESCRIPTION
**Pull request by liberty234**
*  added a new public and private function with export package
*   used sorting as exported package to create public and private function
*  created a new public and private and use d it in the main with network package
* Added a parsing package and used Parsing as a public and divide as a private function
*  added a buffers package which use as public and private function
* Added a new public function and used it in the main and also a …
*  Added a new public function Goods and used it in the main and also a private function with deck package
*  Added a new public function Goods and used it in the main and also a …
*  Added a new public function Goods and used it in the main and also a private function with parsres package
* created a new export package fileops
*  added a public and private function with a new string export package
*   added a public and private function with a new test export package
*  added a public and private function with a new data export package
* added a new public and private function with math package
* update a config package and add a new public and private function
* created a auth package and add a new public and private function
* created a colors package and used colors and colour as public and private function
* used validation as a package and create a variable and also public and private function
* used datetime as a package and create a variable and funcions with  public and private function
* created a new export package logs which i used as public and private function and variable
* created a new export package subs which I used as public and private function and variable
* used csv as an export package which I used as public and private function and variable
* used variable and function as public and private function with an exported cryption package
* used variable and function as public and private function with an exported score package
* used regexops as exported package with private and public
* used files as exported package with private and public
*added a state export package and used variable and function in public and private
* statistics export package and used variable and function in public and private
* used an compression export package and also used variable as  public and private funtion
* used an xml export package and also used variable as  public and private funtion
* created an operation  export package and  used variable as  public and private funtion
* created a tags as an  export package and used variable and function in public and private
* created a email as an  export package and used variable and function in public and private
* created a booking as an  export package and used variable and function in public and private
* created a json as an  export package and used variable and function in public and private
* created a concurrent as an  export package and used variable and function in public and private
* created a authetication as   package and add a new public and private function
* created a webs package which I used as public and private functio
* created a saver package which I used as public and private function
* created a http package which I used as public and private function
* created a databases package which I used as public and private function
* created a variable and function with databases export package
* created a variable and function with calculate export package
* created a variable and function with calculate export package
* created a variable and function with imageops export package
* created a variable and function with jsonops export package
* created a variable and function with jp export package
* Add a structure with public and private fields
*Add a sort with public and private fields
*Add a helpers with public and private fields and added a variable to parser
*Add a dp with public and private fields 
*used xmlops as an exported package with public and private form
*used saverops as an exported package with public and private form

### CODE REVIEW

File: tour-of-go/exports/calulate/calculate.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package calulate
+
+import "fmt"
+
+var Cal = 67 + 98
```
Comment (irediaes) [inline tour-of-go/exports/calulate/calculate.go]: When creating a global variable make sure you add the `type`

File: tour-of-go/exports/calulate/calculate_parser.go [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package calulate
+
+import "fmt"
+
+var par = 83 / 4
```
Comment (irediaes) [inline tour-of-go/exports/calulate/calculate_parser.go]: When creating a global variable make sure you add the `type`

File: tour-of-go/exports/colors/colors.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package colors
+
+import "fmt"
+
+var BestColor = "My best color is Blue"
```
Comment (irediaes) [inline tour-of-go/exports/colors/colors.go]: When creating a global variable make sure you add the `type`

File: tour-of-go/exports/colors/colors_type.go [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package colors
+
+import "fmt"
+
+var blue = "I love blue color"
```
Comment (irediaes) [inline tour-of-go/exports/colors/colors_type.go]: When creating a global variable make sure you add the `type` 

There's a lot of this. Please check all your packages and effect the changes.

File: tour-of-go/exports/comprssion/compression_math.go [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package comprssion
+
+import "fmt"
+
+var cam = 928374 * 938643
+
+func campmath() {
```
Comment (irediaes) [inline tour-of-go/exports/comprssion/compression_math.go]: This is a wrong way of naming a function>
According to what I taught you yesterday, please correct this here and in every affected packages.

`campmath()` should be change to `campMath()`


Why?
1. That is two different words. You have to make the two words visible.
2. It is more readable that way.
3. Improves code quality.
4. And it's golang standard for naming functions.

File: tour-of-go/exports/data/data.go [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+import "fmt"
+
+var Happy string = "my name is happy"
+
+func Class() {
+	class12 := class(45, 23)
+	fmt.Println("The number of class1 and class2 = ", class12)
+}
+
+func FirstName() {
```
Comment (irediaes) [inline tour-of-go/exports/data/data.go]: According to what I taught you yesterday, correct this.

Function name must be a verb.

File: tour-of-go/exports/databases/databases.go [hunk 7]
```diff
@@ -0,0 +1,17 @@
+package databases
+
+import "fmt"
+
+var Data = "Data base"
+
+func DataBase() {
```
Comment (irediaes) [inline tour-of-go/exports/databases/databases.go]: According to what I taught you yesterday, correct this.

Function name must be a verb.


Please correct this in all packages.

File: tour-of-go/exports/http/http_helper.go [hunk 8]
```diff
@@ -0,0 +1,14 @@
+package http
+
+import "fmt"
+
+var log = "log in"
+
+func website() {
+	w := []string{"Facebook", "WhatsApp", "tweeter"}
```
Comment (irediaes) [inline tour-of-go/exports/http/http_helper.go]: Correct the spelling of twitter

File: tour-of-go/exports/main.go [hunk 9]
```diff
@@ -0,0 +1,310 @@
...
+	"github.com/liberty234/lessons/tour-of-go/exports/parsing"
+	"github.com/liberty234/lessons/tour-of-go/exports/regexops"
+	"github.com/liberty234/lessons/tour-of-go/exports/saver"
+	"github.com/liberty234/lessons/tour-of-go/exports/saverops"
+	"github.com/liberty234/lessons/tour-of-go/exports/score"
+	"github.com/liberty234/lessons/tour-of-go/exports/sort"
+	"github.com/liberty234/lessons/tour-of-go/exports/sorting"
+	"github.com/liberty234/lessons/tour-of-go/exports/state"
+	"github.com/liberty234/lessons/tour-of-go/exports/statistics"
+	string1 "github.com/liberty234/lessons/tour-of-go/exports/string"
```
Comment (irediaes) [inline tour-of-go/exports/main.go:45]: This is impressive. Good to know you did your research. 👍

**Review by irediaes — CHANGES_REQUESTED** [review]
Please check my comments below and effect the changes on all affected packages

File: tour-of-go/exports/booking/booking_parser.go [hunk 10]
```diff
@@ -2,9 +2,9 @@ package booking
 
 import "fmt"
 
-var booking = "This ticket have been booked"
+var booking string = "This ticket have been booked"
 
-func bookedTicket() {
+func getbookedTicket() {
```
Comment (irediaes) [inline tour-of-go/exports/booking/booking_parser.go]: Correct the this. change it to `getBookedTicket()"

All new words after the first word in a function name must start with capital letter.

File: tour-of-go/exports/config/config.go [hunk 11]
```diff
@@ -4,7 +4,7 @@ import "fmt"
 
 var GameName string = "FiFa 2023"
 
-func Namevex() {
+func FetchNamevet() {
```
Comment (irediaes) [inline tour-of-go/exports/config/config.go]: Correct the this. change it to `fetchNameVex()"

All new words after the first word in a function name must start with capital letter.


Please check all the places in your packages where you made this same mistake and correct them.

File: tour-of-go/exports/helpers/helpers_get.go [hunk 12]
```diff
@@ -2,7 +2,7 @@ package helpers
 
 var totas int = 45 + 84
 
-func total(bread int, beans int, rice int) int {
+func gettotal(bread int, beans int, rice int) int {
```
Comment (irediaes) [inline tour-of-go/exports/helpers/helpers_get.go]: Fix this

File: tour-of-go/exports/main.go [hunk 13]
```diff
@@ -58,253 +58,253 @@ import (
...
-	string1.Info()
+	string1.GetPersonalInfo()
+	string1.GetInfo()
 
 	//use fileops export package
 	fmt.Println("bill paid", fileops.Paybill)
-	fileops.Bill()
-	fileops.Billpaid()
+	fileops.GetBill()
+	fileops.GetBillpaid()
```
Comment (irediaes) [inline tour-of-go/exports/main.go:82]: Fix this

File: tour-of-go/exports/main.go [hunk 14]
```diff
@@ -58,253 +58,253 @@ import (
...
 	fmt.Println("Yuor Name :", csv.YourName)
-	csv.LastName()
-	csv.MiddleName()
+	csv.GetLastName()
+	csv.GetMiddleName()
 
 	//used encryption as exported package
 	fmt.Println("Ans:", encryption.Tans)
-	encryption.TansAns()
-	encryption.Tanfun()
```
Comment (irediaes) [inline tour-of-go/exports/main.go]: Fix this

**Review by irediaes — CHANGES_REQUESTED** [review]
Please check the new comments.
Some function names were not changed properly.

File: tour-of-go/exports/main.go (hunk 14 above)
Comment (liberty234) [reply tour-of-go/exports/main.go]: Done correcting the error

File: tour-of-go/exports/main.go (hunk 13 above)
Comment (liberty234) [reply tour-of-go/exports/main.go:82]: Done

File: tour-of-go/exports/helpers/helpers_get.go (hunk 12 above)
Comment (liberty234) [reply tour-of-go/exports/helpers/helpers_get.go]: Done changing the error from "gettotal" to "getTotal"

File: tour-of-go/exports/config/config.go (hunk 11 above)
Comment (liberty234) [reply tour-of-go/exports/config/config.go]: corrected

File: tour-of-go/exports/booking/booking_parser.go (hunk 10 above)
Comment (liberty234) [reply tour-of-go/exports/booking/booking_parser.go]: Done

File: tour-of-go/exports/calulate/calculate.go (hunk 1 above)
Comment (liberty234) [reply tour-of-go/exports/calulate/calculate.go]: Done

**Review by irediaes — APPROVED** [review]

**Pull request merged**